src/main/rx/eleres.c

```diff
@@ -0,0 +1,854 @@
+#include "drivers/light_led.h"
+#include "drivers/sound_beeper.h"
+#include "drivers/io.h"
+#include "drivers/rx_spi.h"
+
+#include "fc/rc_controls.h"
+#include "io/gimbal.h"
+#include "io/gps.h"
+#include "io/ledstrip.h"
+#include "io/serial.h"
+#include "fc/cli.h"
+#include "io/statusindicator.h"
+#include "io/osd.h"
+
+#include "rx/rx.h"
+#include "rx/msp.h"
+#include "rx/rx_spi.h"
+#include "rx/eleres.h"
+
+#include "telemetry/telemetry.h"
+#include "flight/mixer.h"
+#include "flight/failsafe.h"
+#include "flight/imu.h"
+#include "flight/pid.h"
+
+#include "navigation/navigation.h"
+
+#include "sensors/battery.h"
+#include "sensors/sensors.h"
+#include "sensors/barometer.h"
+
+#include "io/beeper.h"
+#include "drivers/bus_spi.h"
+#include "fc/runtime_config.h"
+
+#include "config/config_master.h"
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+#include "scheduler/scheduler.h"
+
+#ifdef USE_RX_ELERES
+
+
+PG_REGISTER_WITH_RESET_TEMPLATE(eleresConfig_t, eleresConfig, PG_ELERES_CONFIG, 0);
+
+PG_RESET_TEMPLATE(eleresConfig_t, eleresConfig,
+                  .eleres_freq = 435,
+                  .eleres_telemetry_en = 0,
+                  .eleres_telemetry_power = 7,
+                  .eleres_loc_en = 0,
+                  .eleres_loc_power = 7,
+                  .eleres_loc_delay = 240
+                 );
+
+/*****************    eLeReS compatibile reciver on RFM22B module  ********************/
+rx_spi_received_e Rfm_IRQ(void);
+static uint8_t red_led=0;
+#define RED_LED_ON {if (!red_led) {rfm_spi_write(0x0e, 0x04);red_led=1;}}
+#define RED_LED_OFF {if (red_led) {rfm_spi_write(0x0e, 0x00);red_led=0;}}
+
+#define RC_CHANS 12
+
+static uint8_t hopping_channel = 1;
+
+//int16_t eleresData[RC_CHANS] = { 1500, 1500, 1500, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000};
+
+void RF22B_init_parameter(void);
+void to_rx_mode(void);
+void ChannelHopping(uint8_t hops);
+void telemetry_RX(void);
+void rx_reset(void);
+void to_tx_mode(uint8_t bytes_to_send);
+void to_ready_mode(void);
+void frequency_configurator(uint32_t frequency);
+
+static timeMs_t last_pack_time, next_pack_time, localizer_time;
+static uint8_t DataReady, ch_hopping_time, first_run = 1, loc_force=0;
+static uint8_t hop_list[16];
+uint8_t RSSI = 0, QUALITY;
+static uint16_t good_frames;
+static volatile uint8_t RF_Mode;
+
+uint8_t BkgLoc_enable = 0;
+uint8_t BkgLoc_chlist;
+uint8_t BkgLoc_buf[3][10];
+uint8_t BkgLoc_cnt;
+
+#define Transmit	1
+#define Transmitted	2
+#define Receive 	4
+#define Received 	8
+#define Preamble	16
+
+#define RF22B_Rx_packet_received_interrupt  0x02
+#define RF22B_PACKET_SENT_INTERRUPT         0x04
+#define RF22B_VALID_PREAMBLE_INTERRUPT      0x40
+#define RF22B_VALID_SYNCWORD_INTERRUPT  		0x80
+#define DATA_PACKAGE_SIZE 22
+#define BIN_OFF_VALUE	1150
+#define BIN_ON_VALUE	1850
+uint8_t RF_Tx_Buffer[10];
+uint8_t RF_Rx_Buffer[DATA_PACKAGE_SIZE];
+
+uint8_t tx_full = 0;
+
+
+uint8_t rfm_spi_read(uint8_t address)
```

Comment:
Local functions should be declared `static`. Also general iNav style is to use `camelCase` (it may not look like that because there is quite a bit of historical code, but `camelCase` is preferred for new code).